src/azure-cli/azure/cli/command_modules/appconfig/_params.py

```diff
@@ -125,6 +126,7 @@ def load_arguments(self, _):
         # bypass cli allowed values limitation
         c.argument('separator', validator=validate_separator, help="Delimiter for flattening the json or yaml file to key-value pairs. Separator will be ignored for property files and feature flags. Supported values: '.', ',', ';', '-', '_', '__', '/', ':' ")
         c.argument('profile', validator=validate_import_profile, arg_type=get_enum_type([ImportExportProfiles.DEFAULT, ImportExportProfiles.KVSET]), help="Import profile to be used for importing the key-values. Options 'depth', 'separator', 'content-type', 'label', 'skip-features' and, 'prefix' are not supported when using '{}' profile.".format(ImportExportProfiles.KVSET))
+        c.argument('strict', validator=validate_strict_import, arg_type=get_three_state_flag(), help="Delete all other key-values in the store with specified prefix and label", is_preview=True)
```

Comment:
>Delete all other key-values in the store with specified prefix and label [](http://example.com/codeflow?start=102&length=72)

I am missing some context here. If strict mode is on, then we will delete all other kvs except for the newly imported ones?

Reply:
That's partially correct. We will delete all other kvs **with specified prefix and label** except newly imported ones.
i.e.

Consider the following kvs are present in the config store - 

| Key | Label | Value |
|----|----|----|
| TestApp:BackgroundColor | label1 | Blue |
| TestApp:BackgroundColor | label2 | Red |
| TestApp:Font | label1 | 50 |
| OtherKey | None | OtherValue |

And the file contains following keyvalues - 
| Key | Label | Value |
|----|----|----|
| TestApp:BackgroundColor | label1 | Green |

When user imports using `strict` mode, and specifies the label `label1` and prefix `TestApp`, then 
following actions will happen - 
1. The value for TestApp:BackgroundColor with label1 will be updated to `Green`.
1. The key TestApp:Font with `label1` will be deleted.
1. Other key values will remain unchanged.

The final result in the configuration store would be - 
| Key | Label | Value |
|----|----|----|
| TestApp:BackgroundColor | label1 | Green |
| TestApp:BackgroundColor | label2 | Red |
| OtherKey | None | OtherValue |

Reply:
Thanks for the clarification. Why we check specific label and prefix. First prefix is not a filter, it will just append to the front of every imported key. So in your example, if the key name in file is ```TestApp:BackgroundColor``` and prefix is ```TestApp```, then the newly imported key will be ```TestAppTestApp:BackgroundColor```. It feels odd to remove other existing kv in the store with same prefix. Or we are targeting some specific scenario?

Reply:
Also what will happen for raw import. Will we do any deletions?

Reply:
> So in your example, if the key name in file is `TestApp:BackgroundColor` and prefix is `TestApp`, then the newly imported key will be `TestAppTestApp:BackgroundColor`.
My bad, let me update the example in the original comment.

> Also what will happen for raw import. Will we do any deletions?
Yes we do deletions in raw import. The difference is that, in the raw import, we will delete all missing kvs.

Reply:
For normal kv pair import, if no prefix is specified? will we delete all missing kvs with the same label? I am curious why prefix somehow related with the strict mode?

Reply:
> if no prefix is specified? will we delete all missing kvs with the same label?

Yes, that's the intended behavior.

> I am curious why prefix somehow related with the strict mode?

We are adding a new feature as strict mode for importing from a file, prefix is an existing feature for importing from a file. Since there is no technical limitation that prevents us to support prefix in the strict mode, I am trying to support it.

Reply:
@shenmuxiaosen we already support the concept of strict mode in our [Push Task](https://docs.microsoft.com/en-us/azure/azure-app-configuration/push-kv-devops-pipeline#parameters) and [Github Actions](https://docs.microsoft.com/en-us/azure/azure-app-configuration/concept-github-action#use-strict-sync). They both support filtering by prefix and label. That's why we want to do the same in CLI.

Basically, customers are encouraged to organize their key-values in a way that they can filter by prefix and/or label. In strict mode, we are offering the same filtering capability.